rllib/algorithms/a3c/a3c.py

```diff
@@ -214,33 +202,30 @@ def sample_and_compute_grads(worker: RolloutWorker) -> Dict[str, Any]:
         with self._timers[GRAD_WAIT_TIMER]:
             # Results are a mapping from ActorHandle (RolloutWorker) to their
             # returned gradient calculation results.
-            self._worker_manager.call_on_all_available(sample_and_compute_grads)
-            async_results = self._worker_manager.get_ready()
+            self.workers.foreach_worker_async(
+                func=sample_and_compute_grads,
+            )
+            async_results = self.workers.fetch_ready_async_reqs()
 
         # Loop through all fetched worker-computed gradients (if any)
         # and apply them - one by one - to the local worker's model.
         # After each apply step (one step per worker that returned some gradients),
         # update that particular worker's weights.
         global_vars = None
         learner_info_builder = LearnerInfoBuilder(num_devices=1)
-        for worker, results in async_results.items():
-            for result in results:
-                # Apply gradients to local worker.
-                with self._timers[APPLY_GRADS_TIMER]:
-                    local_worker.apply_gradients(result["grads"])
-                self._timers[APPLY_GRADS_TIMER].push_units_processed(
-                    result["agent_steps"]
-                )
-
-                # Update all step counters.
-                self._counters[NUM_AGENT_STEPS_SAMPLED] += result["agent_steps"]
-                self._counters[NUM_ENV_STEPS_SAMPLED] += result["env_steps"]
-                self._counters[NUM_AGENT_STEPS_TRAINED] += result["agent_steps"]
-                self._counters[NUM_ENV_STEPS_TRAINED] += result["env_steps"]
-
-                learner_info_builder.add_learn_on_batch_results_multi_agent(
-                    result["infos"]
-                )
+        for actor_id, result in async_results:
```

Comment:
I think the typedef for the return value of the WorkerSet fetch_asyn_reqs is wrong.
It should be `List[Tuple[int, T]]`, correct?

Reply:
yes, good catch.